cylc/flow/data_store_mgr.py

```diff
@@ -698,7 +698,10 @@ def _expand_graph_window(
             self, s_id, s_node, items, active_id, flow_label, reflow,
             edge_distance, descendant=False, is_parent=False):
         """Construct nodes/edges for children/parents of source node."""
+        final_point = self.schd.config.final_point
         for t_name, t_point, _ in items:
+            if t_point > final_point:
+                continue
```

Comment:
I was going to say the `final_point` local variable is unnecessary here, but I guess there would be a small performance hit from looking up `self.schd.config` attributes inside the loop :thinking:

Reply:
Yeah, I'm not certain what's fastest.. I suppose I could do a test to find out.

Reply:
I googled it, it is faster to look up a local variable than nested attributes. It just requires more low-level ops. Inside a loop that can matter.  In this case I doubt it'll be noticeable either way.

Reply:
Yeah, how many children can a node have 100? 1000?

Reply:
Let's leave it as-is, just in case.

Reply:
This way is better for performance :+1:.